pkg/distro/defs/loader.go

```diff
@@ -349,3 +362,38 @@ func load(distroNameVer string) (*toplevelYAML, error) {
 
 	return &toplevel, nil
 }
+
+// ImageConfig returns the image type specific ImageConfig
+func ImageConfig(distroNameVer, archName, typeName string, replacements map[string]string) (*distro.ImageConfig, error) {
+	toplevel, err := load(distroNameVer)
+	if err != nil {
+		return nil, err
+	}
+	typeName = strings.ReplaceAll(typeName, "-", "_")
```

Comment:
Will this cause confusion, the mismatch between the name in the yaml and how it appears in the CLI and API?  Having names-with-dashes is possible with quotes, so maybe we could make it a habit or convention to define image type names that way?  Or would that be too ugly?

What I'm worried about here is that, if I was a user defining my own image type externally, I might name it "my-image-type" in the yaml, and then trying to `ib build my-image-type` wouldn't work.

Reply:
Another idea might be to make image types an array in yaml with a `name` parameter, just like we do internally, and build the map as we load.

Reply:
Adding quotes around the map seems okay to me, I did a commit with that and pushed.